File: java/transpose/src/main/java/Transpose.java
public class Transpose {
    public String transpose(String toTranspose) {
        // Handle empty string case
        if (toTranspose.isEmpty()) {
            return "";
        }

        // Split the input into rows
        String[] rows = toTranspose.split("\n");
        
        // Find the maximum row length
        int maxRowLength = 0;
        for (String row : rows) {
            maxRowLength = Math.max(maxRowLength, row.length());
        }
        
        // Build the transposed output
        StringBuilder result = new StringBuilder();
        
        for (int col = 0; col < maxRowLength; col++) {
            for (int row = 0; row < rows.length; row++) {
                // If this column exists in the current row, add the character
                // Otherwise, add a space if there are non-space characters to the right
                if (col < rows[row].length()) {
                    result.append(rows[row].charAt(col));
                } else {
                    // Check if we need to add a space (if there are characters to the right in rows below)
                    boolean needSpace = false;
                    for (int checkRow = row + 1; checkRow < rows.length; checkRow++) {
                        if (col < rows[checkRow].length()) {
                            needSpace = true;
                            break;
                        }
                    }
                    if (needSpace) {
                        result.append(' ');
                    }
                }
            }
            
            // Add a newline if this is not the last column
            if (col < maxRowLength - 1) {
                result.append('\n');
            }
        }
        
        return result.toString();
    }
}

File: python/transpose/transpose.py
def transpose(text):
    # Handle empty string case
    if not text:
        return ""
    
    # Split the input text into lines
    lines = text.split('\n')
    
    # Find the maximum line length
    max_length = max(len(line) for line in lines)
    
    # Initialize the result
    result = []
    
    # Transpose the matrix
    for i in range(max_length):
        new_line = []
        for j in range(len(lines)):
            # If the current position is within the line's length, add the character
            # Otherwise, add a space if it's not the rightmost column that needs padding
            if i < len(lines[j]):
                new_line.append(lines[j][i])
            else:
                # Only add spaces for positions that have characters below them
                # This implements the "pad to the left with spaces" rule
                should_pad = False
                for k in range(j + 1, len(lines)):
                    if i < len(lines[k]):
                        should_pad = True
                        break
                
                if should_pad:
                    new_line.append(' ')
                else:
                    # Don't add anything if there are no characters below
                    # This implements the "don't pad to the right" rule
                    break
        
        result.append(''.join(new_line))
    
    # Join the transposed lines with newlines
    return '\n'.join(result)
